### PR TITLE
Update Explorer URLs due to rebranding

### LINKED ISSUE #914: Rebranding: Update Explorer URLs in Explorer
**Description**

Update URLs used on explorer  to self identify its own stages: https://github.com/gnosis/gp-ui/blob/6620910e66f573684de86f914f456b0fff87cd21/webpack.config.js#L23-L26

**Proposal**

New prod URL: `explorer.cow.fi`
New barn URL: `barn.explorer.cow.fi`
*New staging URL: `staging.explorer.cow.fi`
*New staging URL: `dev.explorer.cow.fi`

* Still needs to confirm this will be the final domain

**Additional context**

### PR DESCRIPTION
**Pull request by henrypalacios**
Closes #914, #915

- Update URLs to identify PROD, BARN stages with base `explorer.cow.fi`

- Update URLs to talk to backend API with base `api.cow.fi`

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -22,15 +22,15 @@ const EXPLORER_APP = {
...
+    EXPLORER_APP_DOMAIN_REGEX_PROD: '^explorer\\.cow\\.fi',
+    EXPLORER_APP_DOMAIN_REGEX_BARN: '^barn\\.explorer\\.cow\\.fi',
 
-    OPERATOR_URL_STAGING_MAINNET: 'https://protocol-mainnet.dev.gnosisdev.com/api',
-    OPERATOR_URL_STAGING_RINKEBY: 'https://protocol-rinkeby.dev.gnosisdev.com/api',
-    OPERATOR_URL_STAGING_XDAI: 'https://protocol-xdai.dev.gnosisdev.com/api',
-    OPERATOR_URL_PROD_MAINNET: 'https://protocol-mainnet.gnosis.io/api',
-    OPERATOR_URL_PROD_RINKEBY: 'https://protocol-rinkeby.gnosis.io/api',
-    OPERATOR_URL_PROD_XDAI: 'https://protocol-xdai.gnosis.io/api',
+    OPERATOR_URL_STAGING_MAINNET: 'https://bar.api.cow.fi/mainnet',
```
Comment (alongoni) [inline webpack.config.js:28]: should be `barn` 🙃

**Review by alongoni — CHANGES_REQUESTED** [review]

**Review by alongoni — APPROVED** [review]

**Comment by henrypalacios** [comment]
I have only one doubt here. According to the statement in #915 the backend API path will remain with the name `xdai` network.
I also see the same in:
https://github.com/gnosis/cowswap/pull/2047

Is it OK?

File: webpack.config.js [hunk 2]
```diff
@@ -22,15 +22,15 @@ const EXPLORER_APP = {
...
 
-    OPERATOR_URL_STAGING_MAINNET: 'https://protocol-mainnet.dev.gnosisdev.com/api',
-    OPERATOR_URL_STAGING_RINKEBY: 'https://protocol-rinkeby.dev.gnosisdev.com/api',
-    OPERATOR_URL_STAGING_XDAI: 'https://protocol-xdai.dev.gnosisdev.com/api',
-    OPERATOR_URL_PROD_MAINNET: 'https://protocol-mainnet.gnosis.io/api',
-    OPERATOR_URL_PROD_RINKEBY: 'https://protocol-rinkeby.gnosis.io/api',
-    OPERATOR_URL_PROD_XDAI: 'https://protocol-xdai.gnosis.io/api',
+    OPERATOR_URL_STAGING_MAINNET: 'https://barn.api.cow.fi/mainnet',
+    OPERATOR_URL_STAGING_RINKEBY: 'https://barn.api.cow.fi/rinkeby',
+    OPERATOR_URL_STAGING_XDAI: 'https://barn.api.cow.fi/xdai',
```
Comment (alfetopito) [inline webpack.config.js:30]: Barn api URLs are not yet ready and we'll release without them, can you please revert this change?

File: webpack.config.js [hunk 3]
```diff
@@ -22,15 +22,15 @@ const EXPLORER_APP = {
   envVars: {
     EXPLORER_APP_DOMAIN_REGEX_DEV: '^protocol-explorer\\.dev|^localhost:\\d{2,5}|^pr\\d+--gpui\\.review',
     EXPLORER_APP_DOMAIN_REGEX_STAGING: '^protocol-explorer\\.staging',
-    EXPLORER_APP_DOMAIN_REGEX_PROD: '^gnosis-protocol\\.io',
-    EXPLORER_APP_DOMAIN_REGEX_BARN: '^barn\\.gnosis-protocol\\.io',
+    EXPLORER_APP_DOMAIN_REGEX_PROD: '^explorer\\.cow\\.fi',
+    EXPLORER_APP_DOMAIN_REGEX_BARN: '^barn\\.explorer\\.cow\\.fi',
```
Comment (alfetopito) [inline webpack.config.js:26]: One more change, can you make the regexes accept both old and new domains?
E.g.: `^(explorer\\.cow\\.fi|gnosis-protocol\\.io)`

For the transition period would be better to have them both working

**Comment by alfetopito** [comment]
> I have only one doubt here. According to the statement in #915 the backend API path will remain with the name `xdai` network. I also see the same in: [gnosis/cowswap#2047](https://github.com/gnosis/cowswap/pull/2047)
> 
> Is it OK?

Missed this question.
Yes, it'll remain `xdai` for now

File: webpack.config.js (hunk 2 above)
Comment (henrypalacios) [reply webpack.config.js]: Done!

File: webpack.config.js (hunk 3 above)
Comment (henrypalacios) [reply webpack.config.js]: Done! 

What I'm not sure is if **barn** will be available for `EXPLORER_APP_DOMAIN_REGEX_BARN`  like as `barn.explorer.cow.fi`

Comment (alfetopito) [reply webpack.config.js]: Yes, barn is available for the explorer domain.
It just isn't yet for the api domains.

File: webpack.config.js [hunk 4]
```diff
@@ -22,15 +22,15 @@ const EXPLORER_APP = {
...
 
     OPERATOR_URL_STAGING_MAINNET: 'https://protocol-mainnet.dev.gnosisdev.com/api',
     OPERATOR_URL_STAGING_RINKEBY: 'https://protocol-rinkeby.dev.gnosisdev.com/api',
     OPERATOR_URL_STAGING_XDAI: 'https://protocol-xdai.dev.gnosisdev.com/api',
-    OPERATOR_URL_PROD_MAINNET: 'https://protocol-mainnet.gnosis.io/api',
-    OPERATOR_URL_PROD_RINKEBY: 'https://protocol-rinkeby.gnosis.io/api',
-    OPERATOR_URL_PROD_XDAI: 'https://protocol-xdai.gnosis.io/api',
+    OPERATOR_URL_PROD_MAINNET: 'https://api.cow.fi/mainnet',
+    OPERATOR_URL_PROD_RINKEBY: 'https://api.cow.fi/rinkeby',
+    OPERATOR_URL_PROD_XDAI: 'https://api.cow.fi/xdai',
```
Comment (alfetopito) [inline webpack.config.js:33]: Ah, another thing I failed to tell you.

We wanted to remove the suffix `/api` but we won't for now as it'll take longer on the backend side.
Please add it to them like it was with the previous domain:

```suggestion
    OPERATOR_URL_PROD_MAINNET: 'https://api.cow.fi/mainnet/api',
    OPERATOR_URL_PROD_RINKEBY: 'https://api.cow.fi/rinkeby/api',
    OPERATOR_URL_PROD_XDAI: 'https://api.cow.fi/xdai/api',
```

**Review by alfetopito — APPROVED** [review]
